test/scripts/deploy.test.ts

```diff
@@ -0,0 +1,138 @@
+import { expect, use } from 'chai';
+import chaiAsPromised from 'chai-as-promised';
+import { Contract, ContractFactory } from 'ethers';
+import fs from 'fs';
+import * as hardhat from 'hardhat';
+import { ethers } from 'hardhat';
+import sinon from 'sinon';
+import {
+  ContractAddresses,
+  deployContracts,
+  generateJsonConfig,
+} from '../../scripts/deploy';
```

Comment:
Wouldn't the main() function be called upon doing an import on scripts/deploy.ts? Or am I misunderstanding how JS/TS modules work?

Reply:
aaah....hm...good point, that didn't occur to me. Though, I'm not sure if it actually affects anything much, since you are only going to run unit tests when developing and in CI.
At some point you will have to import the main anyway in order to test it, won't you?

Reply:
Mmm this is a difficult topic I'd say. For unit testing I would definitely look for a way to not run main() if it is possible. Even for the CI process, running the main() function means more resources used (since we are running actual code).  Now imagine if we had multiple deploy scripts :(

Reply:
I've added a env var TEST=true to npm test script.